### PR TITLE
#184797812  admin should able to crud on users

### PR DESCRIPTION
**Pull request by edwinnambaje**
#### What does this PR do?

-Adds Crud operations to an Admin user

#### Description of Task to be completed?
- As an admin user ,I should be able to run CRUD(create, update and delete) operations against users model

#### How should this be manually tested?
after cloning the repo,
- in terminal add `npm install`
- add `.env` in your repo and assign the variables as there are in `.env.example` file
- run in terminal `npm run migrate:all` to setup the database
- run in terminal `npm start` to start the project
- in your browser add `http://localhost:<your port>/api-docs` for documentation
- Navigate to the admin routes 

#### Any background context you want to provide?
-N/A

#### What are the relevant pivotal tracker/Trello stories?
-https://www.pivotaltracker.com/n/projects/2632830/stories/184797812

#### Screenshots (if appropriate)
Screenshot 2023-03-29 at 01 04 31
Screenshot 2023-03-29 at 01 05 18
Screenshot 2023-03-29 at 01 06 04
Screenshot 2023-03-29 at 01 17 31
Screenshot 2023-03-29 at 01 09 02
Screenshot 2023-03-29 at 01 10 35
Screenshot 2023-03-29 at 01 10 56
Screenshot 2023-03-29 at 01 11 25
Screenshot 2023-03-29 at 01 11 33
Screenshot 2023-03-29 at 01 27 35

### CODE REVIEW

**Review by Zeden12 — APPROVED** [review]
LGTM

**Review by Patienceineza — APPROVED** [review]
LGTM

File: events/eventHandler/user.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  try {
+    const { olduser, token } = data;
+    await new SendEmail(
+      olduser,
+      `${process.env.UI_URL}/users/password-reset/${token}`,
+      null
+    ).reset();
+  } catch (error) {
+    /* istanbul ignore next */
+    console.log(error);
```
Comment (nshutijonathan) [inline events/eventHandler/user.js]: return error

File: events/eventHandler/user.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  }
+});
+
+userEventEmitter.on('disableAccount', async (data) => {
+  try {
+    const { user } = data;
+    await new SendEmail(user, null, null).deactivate();
+  } catch (error) {
+    console.log(error);
```
Comment (nshutijonathan) [inline events/eventHandler/user.js]: remove console.log and return error

File: events/eventHandler/user.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+  }
+});
+
+userEventEmitter.on('activateAccount', async (data) => {
+  /* istanbul ignore next */
+  try {
+    const { user } = data;
+    await new SendEmail(user, null, null).activate();
+  } catch (error) {
+    console.log(error);
```
Comment (nshutijonathan) [inline events/eventHandler/user.js]: remove console and return error

File: events/eventHandler/user.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+  }
+});
+
+userEventEmitter.on('twoFactorAuth', async (data) => {
+  try {
+    const { userInfo, randomAuth } = data;
+    await new SendEmail(userInfo, null, randomAuth).twoFactorAuth();
+  } catch (error) {
+    console.log(error);
```
Comment (nshutijonathan) [inline events/eventHandler/user.js]: remove console.log

File: src/controllers/admin.controller.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+      if (email) user.email = email;
+      if (username) user.username = username;
+      if (password) user.password = await bcrypt.hash(password, saltRounds);
+      if (gender) user.gender = gender;
+      if (role) user.role = role;
+      if (verified) user.verified = verified;
+      await user.save();
+      return res.json({ data: user });
+    } catch (err) {
+      console.error(err);
```
Comment (nshutijonathan) [inline src/controllers/admin.controller.js]: remove `console.log`

**Review by nshutijonathan — COMMENTED** [review]
left some comments

**Comment by edwinnambaje** [comment]
> left some comments

i have resolved them

**Review by Habihirwe — APPROVED** [review]
well done

**Review by nshutijonathan — APPROVED** [review]
LGTM

**Review by UmuhireJessie — APPROVED** [review]

**Pull request merged**